Meta/Lagom/Contrib/VideoPlayerSDL/main.cpp

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2023, Stephan Vedder <stephan.vedder@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibCore/ArgsParser.h>
+#include <LibCore/EventLoop.h>
+#include <LibCore/File.h>
+#include <LibCore/MappedFile.h>
+#include <LibMain/Main.h>
+#include <LibVideo/PlaybackManager.h>
+#include <SDL2/SDL.h>
+
+ErrorOr<int> serenity_main(Main::Arguments arguments)
+{
+    StringView filename = ""sv;
+    Core::ArgsParser args_parser;
+    args_parser.add_positional_argument(filename, "The video file to display.", "filename", Core::ArgsParser::Required::No);
+    args_parser.parse(arguments);
+
+    if (filename.is_empty()) {
+        warnln("No filename given");
+        return 1;
+    }
+
+    auto file = TRY(Core::File::open(filename, Core::File::OpenMode::Read));
+    auto mapped_file = TRY(Core::MappedFile::map_from_file(move(file), filename));
+    auto load_file_result = Video::PlaybackManager::from_mapped_file(move(mapped_file));
+    if (load_file_result.is_error()) {
+        warnln("Failed to decode file {}", filename);
+        return 1;
+    }
+
+    SDL_Texture* texture = NULL;
+
+    auto playback_manager = load_file_result.release_value();
+    playback_manager->on_video_frame = [&texture](RefPtr<Gfx::Bitmap> frame) {
+        u8* pixels = frame->scanline_u8(0);
+        int result = SDL_UpdateTexture(texture, NULL, pixels, frame->pitch());
```

Comment:
If `SDL_UpdateTexture` expects a buffer sized to `w * h * 4` bytes (if that's the size of the pixel format), then this will overread if the current frame doesn't match the initial video resolution.

Note that the `Track::video_data()` is not really accurate, the values there can change at any time, including on the first decoded frame.

To avoid this, I would create/recreate the texture inside `on_video_frame` every time the video resolution changes. I suppose that if the texture is copied 1:1 to the window, you'll have to resize the window at the same time.